### ISSUE TITLE
[Bug]: ReactiveUI.Fody prevents hot reload on net7

### ISSUE THREAD

**Issue by Yarith** [reactiveui/ReactiveUI#3432]
### Describe the bug 🐞

If you have upgraded the project to .net7 and are using the ReactiveUI.Fody package, you lose the ability to use hot reload and edit and continue in this project.

German error text:
```
Fehler	CS7038	Fehler beim Ausgeben von Modul "ReactiveUIHotReload": Das Ändern der Version eines Assemblyverweises ist während des Debuggens nicht zulässig: "System.Runtime, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" hat die Version in "7.0.0.0" geändert.	ReactiveUIHotReload		1	Aktiv
```

English error text:
```
CS7038: Failed to emit module 'ReactiveUIHotReload': Changing the version of an assembly reference is not allowed during debugging:
'System.Runtime, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' changed version to '7.0.0.0'.
```

### Step to reproduce

1. Create a fresh net7 wpf project via: `dotnet new wpf`
2. Add the `ReactiveUI.Fody` package
3. Create a viewmodel which implements `ReactiveObject` and uses the `[Reactive]` attribute:
```csharp
public sealed class MainViewModel : ReactiveObject
{
    [Reactive]
    public required string Text { get; set; }
}
```
4. Run the project
5. Change something small in the project, like adding a `MessageBox.Show("Test")` below the `InitializeComponent()` in the `MainWindow` code behind. 
6. Press on the flame icon to apply code changes.
7. It fails with the compile error above.

### Reproduction repository

https://github.com/reactiveui/ReactiveUI

### Expected behavior

ReactiveUI.Fody should not interfere with hot reload and edit and continue on .net 7 projects.

### IDE

Visual Studio 2022

### ReactiveUI Version

15,3.1, 15.4.1. It seems to affect all versions of ReactiveUI.Fody.

### Additional information ℹ️

I was plagued in the .net 6 time frame with this error, but i didn't knew that `ReactiveUI.Fody` was the reason. As soon as i updated it to the latest version, the problem went away. Based on the information on nuget.org it seems that it went away as soon as the package targeted `net6`. I don't know when ReactiveUI will target `net7` so i have filled this issue. Maybe there is a solution without changing the target .net version of the package?

**Comment by glennawatson** [comment]
We will target net7 eventually. There's been some build issues which should be fixed by net 7.0.200

**Comment by ChrisPulman** [comment]
Thank you for this, we are currently waiting for a bug fix in Net 7.0 to hopefully allow us to target Net 7.0. We are tracking the error in the DotNet project and hope they will resolve the issues with their next release.

**Comment by Yarith** [comment]
I can confirm that the current version 18.4.26 does not prevent debugging or hot reload in Visual Studio on my .net7 projects with ReactiveUI.Fody anymore. I expect this issue will stay resolved until the update to .net8 breaks it again.
